zio/detector/file.go

```diff
@@ -10,16 +10,22 @@ import (
 )
 
 func OpenFile(zctx *resolver.Context, path string, flags *zio.ReaderFlags) (*zbuf.File, error) {
-	info, err := os.Stat(path)
-	if err != nil {
-		return nil, err
-	}
-	if info.IsDir() {
-		return nil, errors.New("is a directory")
-	}
-	f, err := os.Open(path)
-	if err != nil {
-		return nil, err
+	var f *os.File
+	var err error
+	if path == "-" {
```

Comment:
Shouldn't the "translate '-' to /dev/stdin" step stay in zq?  I don't think we want zq to try reading from stdin if passed a file named "-". 

(I've wanted to do this little bit of factoring for a while and this is what prevented me from doing it.)

Reply:
One option would be to have this func take a `[]os.File` or `[]io.Reader` instead.

Reply:
Hmm, I'm not worried about a file named "-" and it's pretty convenient to have this here.  What if we used path="" to mean stdin and the caller could check for "-"?